Makefile

```diff
@@ -26,7 +26,7 @@ __check_defined = \
     $(if $(value $1),, \
       $(error Undefined $1$(if $2, ($2))))
 
-all: tools build test fmtcheck validate_commits vet image ## Compiles binary and runs format and style checks
+all: tools build test fmtcheck validate_commits vet image lint ## Compiles binary and runs format and style checks
```

Comment:
+1

Reply:
Let's add something like

    LINT_PACKAGES = $(shell echo $(PACKAGES) | sed -e 's@github.com/fabric8-services/fabric8-jenkins-idler/internal/configuration@@')

under the PACKAGES definition above. And then use this variable in 'lint'

Reply:
The rub is really that golint does not provide any mechanism to prevent a check. These are really suggestions as mentioned in the README. On one side I really would like to be as close as possible to the suggestions, but it comes at a cost. 

Problem with not failing a build when there are violations is that they will eventually sneak in. For now, I'd go with the change to the Makefile. I find most of the changes made as part of this pull request reasonable otherwise. One can discuss changes in policies and enforcement of rules when another potential conflict arises. @
kishansagathiya, does tha make sense?

Reply:
BTW, please make 'all' phony as well. Just a goos measure

Reply:
Makes sense. 
Since we lint would apply only on listed package, we would be able to add ALL_CAPS variable names as we have in config

Reply:
Doesn't it make sense to keep all targets as phony?

Reply:
no, $(BUILD_DIR), $(BUILD_DIR)/$(REGISTRY_IMAGE), image, tools are all file based and are not phony